src/Compilers/CSharp/Portable/Lowering/LambdaRewriter/LambdaRewriter.cs

```diff
@@ -341,7 +352,8 @@ private void MakeFrames(ArrayBuilder<ClosureDebugInfo> closureDebugInfo)
 
                 foreach (var captured in capturedVars)
                 {
-                    if (!_analysis.VariableScope.TryGetValue(captured, out BoundNode captureScope))
+                    var declarationScope = Analysis.GetVariableDeclarationScope(scope, captured);
+                    if (declarationScope == null)
```

Comment:
Nope, this is valid. It means that the captured variable lives in the top-level scope, i.e. the class scope. I'm have rewritten this method entirely in a later PR so I don't see any need in adding extra validation here.